src/mqtt.c

```diff
@@ -75,7 +76,10 @@ struct mqtt_client_conf {
   /* For publishing */
   char *topic_prefix;
   _Bool store_rates;
+  _Bool store_timestamp;
   _Bool retain;
+  char *pattern;
```

Comment:
There's no need to keep this string around – it's not read outside of the config function, which stores it. Instead, make this a local variable in the config callback.